integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -351,11 +351,32 @@ def test_floor(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).selectExpr('floor(a)'))
 
+@pytest.mark.skipif(is_before_spark_330(), reason='scale parameter in Floor function is not supported before Spark 3.3.0')
+@pytest.mark.parametrize('data_gen', double_n_long_gens + _arith_decimal_gens_no_neg_scale, ids=idfn)
+def test_floor_scale_zero(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).selectExpr('floor(a, 0)'),
+            conf={'spark.rapids.sql.castFloatToDecimal.enabled':'true'})
+
 @pytest.mark.parametrize('data_gen', double_n_long_gens + _arith_decimal_gens_no_neg_scale, ids=idfn)
 def test_ceil(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).selectExpr('ceil(a)'))
 
+@pytest.mark.skipif(is_before_spark_330(), reason='scale parameter in Ceil function is not supported before Spark 3.3.0')
+@pytest.mark.parametrize('data_gen', double_n_long_gens + _arith_decimal_gens_no_neg_scale, ids=idfn)
+def test_ceil_scale_zero(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).selectExpr('ceil(a, 0)'),
+            conf={'spark.rapids.sql.castFloatToDecimal.enabled':'true'})
+
+@pytest.mark.skipif(is_before_spark_330(), reason='scale parameter in Ceil function is not supported before Spark 3.3.0')
+@allow_non_gpu('ProjectExec')
+@pytest.mark.parametrize('data_gen', double_n_long_gens + _arith_decimal_gens_no_neg_scale, ids=idfn)
+def test_ceil_scale_nonzero(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).selectExpr('ceil(a, -1)'))
```

Comment:
would it be an overkill to ask for a int_gen for the scale value even if we limit it to -5, 5 range?

Reply:
Sorry, I did not understand your question completely. 
Cudf doesn't support `scale` for Ceil/Floor functions. Those are currently unaryExpressions. So until we have cudf implementation of scale for these functions plan is to fall back to CPU. Scale with value `0` is an exception as it is same as calling Ceil/Floor functions. Just that the physical plan changes.

Reply:
Then what are we really testing in this test? Would it make sense to add this test when we actually support ceil/floor with scale?

Reply:
Had added this test for completion i.e to show that we fall back to cpu for non zero scale by having allow_non_gpu. Your suggestion makes sense. I can remove this test and just have scale=0 test.

Reply:
Removed this test. PTAL.